lib/kv-redis.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var assert = require('assert');
+var Connector = require('loopback-connector').Connector;
+var debug = require('debug')('loopback:connector:redis');
+var Redis = require('ioredis');
+var Command = Redis.Command;
+var util = require('util');
+
+exports.initialize = function initializeDataSource(dataSource, callback) {
+  var settings = dataSource.settings;
+
+  // TODO handle settings.lazyConnect - use ioredis option "lazyConnect"
+  dataSource.connector = new RedisKeyValueConnector(settings);
+
+  if (!callback) return;
```

Comment:
Good catch, will fix by listening for `connect` and `error` events provided by Redis connector.